File: qe_selenium_java/Activity8.java
package demos;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;

public class Activity8 {

    public static void main(String[] args) throws InterruptedException {

        // If geckodriver is not in PATH, uncomment and set path
        // System.setProperty("webdriver.gecko.driver", "path/to/geckodriver.exe");

        WebDriver driver = new FirefoxDriver();
        driver.manage().window().maximize();

        try {
            // Open site
            driver.get("https://training-support.net/webelements/mouse-events");
            Thread.sleep(2000);

            System.out.println("Page title: " + driver.getTitle());

            Actions actions = new Actions(driver);

            // Click Cargo.lock -> Cargo.toml
            WebElement cargoLock = driver.findElement(By.id("cargo-lock"));
            WebElement cargoToml = driver.findElement(By.id("cargo-toml"));

            actions.click(cargoLock)
                    .moveToElement(cargoToml)
                    .click()
                    .build()
                    .perform();

            Thread.sleep(1000);
            System.out.println("Confirmation: " +
                    driver.findElement(By.id("result")).getText());

            // Double click src and right click target
            WebElement src = driver.findElement(By.id("src"));
            WebElement target = driver.findElement(By.id("target"));

            actions.doubleClick(src)
                    .contextClick(target)
                    .build()
                    .perform();

            Thread.sleep(1000);

            // Click Open
            WebElement openOption = driver.findElement(By.id("open"));
            openOption.click();

            Thread.sleep(1000);
            System.out.println("Confirmation: " +
                    driver.findElement(By.id("result")).getText());

        } finally {
            driver.quit();
        }
    }
}

File: qe_selenuim_python/Activity8.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.action_chains import ActionChains
import time

driver = webdriver.Firefox()
driver.maximize_window()

try:
    driver.get("https://training-support.net/webelements/mouse-events")
    time.sleep(2)

    print("Page title:", driver.title)

    actions = ActionChains(driver)

    cargo_lock = driver.find_element(By.ID, "cargo-lock")
    cargo_toml = driver.find_element(By.ID, "cargo-toml")

    actions.click(cargo_lock)\
           .move_to_element(cargo_toml)\
           .click()\
           .perform()

    time.sleep(1)
    print("Confirmation:", driver.find_element(By.ID, "result").text)

    src = driver.find_element(By.ID, "src")
    target = driver.find_element(By.ID, "target")

    actions.double_click(src)\
           .context_click(target)\
           .perform()

    time.sleep(1)

    open_option = driver.find_element(By.ID, "open")
    open_option.click()

    time.sleep(1)
    print("Confirmation:", driver.find_element(By.ID, "result").text)

finally:
    driver.quit()
